File: e179/java/Solution.java
class Solution {
    /**
     *@param n, m: Two integer
     *@param i, j: Two bit positions
     *return: An integer
     */
    public int updateBits(int n, int m, int i, int j) {
        // write your code here
        int a = j+1 >= 32?0:1 << (j+1);
        int tmp = n & ( a - (1 << i) );

        return n - tmp + (m << i);
        
    }
}

File: e179/python/solution.py
class Solution:
    #@param n, m: Two integer
    #@param i, j: Two bit positions
    #return: An integer
    def updateBits(self, n, m, i, j):
        # write your code here

        #tmp = n & (pow(2, j+1) - pow(2, i))
        tmp = n & ((1 << j+1) - (1 << i))
        #print tmp,n
        return self.int_overflow(n - tmp + (m << i))
        
    def int_overflow(self, val):
        maxint = 2147483647  
        if not -maxint-1 <= val <= maxint:  
            val = (val + (maxint + 1)) % (2 * (maxint + 1)) - maxint - 1  
        return val 
